cmd/cmd_utils.go

```diff
@@ -407,6 +409,32 @@ func executeCustomCommand(
 	}
 }
 
+func handleArgsAfterDoubleDash(args []string, osArgs []string) ([]string, string) {
+	doubleDashIndex := lo.IndexOf(osArgs, "--")
+	mainArgs := args
+	doubleDashString := ""
+	if doubleDashIndex > 0 {
+		mainArgs = lo.Slice(osArgs, 0, doubleDashIndex)
+		doubleDashString = strings.Join(lo.Slice(osArgs, doubleDashIndex+1, len(osArgs)), " ")
+		result := []string{}
+		lookup := make(map[string]bool)
+
+		// Populate a lookup map for quick existence check
+		for _, val := range mainArgs {
+			lookup[val] = true
+		}
+
+		// Iterate over leftArr and collect matching elements in order
+		for _, val := range args {
+			if lookup[val] {
+				result = append(result, val)
+			}
+		}
+		mainArgs = result
+	}
+	return mainArgs, doubleDashString
```

Comment:
```suggestion
	return mainArgs, nativeArgs
```

Reply:
Can't we do this simpler:
```
func extractNativeArgs(args []string) ([]string, []string) {
    for i, arg := range args {
        if arg == "--" {
            return args[:i], args[i+1:] // Split before and after "--"
        }
    }
    return args, nil // No "--" found, return args as-is
}
```

Reply:
Unfortunately not because cobra is consuming double dash. So it is no more available in args that cobra passes to Run function. Let me check if somehow i can force cobra to spit double dash.

Reply:
Okay so we would have to drop the flag hint about double dash to do it. I don't think that is a good trade off.  Flag hint helps our customers know about possible arguments.

Reply:
Ok, then at least I understand why we do it this way.